### PR TITLE
embedded vault (encrypted yaml values) 

### PR DESCRIPTION
**Pull request by alikins**
##### ISSUE TYPE
- Feature Pull Request
##### ANSIBLE VERSION

2.2. devel
##### SUMMARY

Part of 2.2 roadmap vault changes "Add ‘per variable’ vault support (!vault YAML directive, existing PR already)"

This is based on https://github.com/h0nIg/ansible/commit/e680f504aa0f6826ff47263d79dfb19d8a74b0e7 mentioned in the thread for https://github.com/ansible/ansible/issues/13287

### CODE REVIEW

**Comment by sivel** [comment]
We had talked about the potential of marking the decrypted data as having come from a vault, and effectively set it as `no_log`.  I'm not sure that is a requirement, but would be a super nice to have.

Something like that may require us to create some form of class that we wrap the data in, similar to Unsafe, so that it is easy to tag that data.

Just some thoughts and ideas.

**Comment by bcoca** [comment]
yes, make it something similar to 'unsafe' and always mark as no_log

**Comment by h0nIg** [comment]
nice to see that my code is used as a base :)

the current code looks like the solution, no need to decrypt to a complex type like a map/list. but it may be necessary to include the vault version, to prevent problems with the encryption on future releases

**Comment by alikins** [comment]
I think the version as of a11742ab0a14d90ada6f94898df27c36e3769388 seems to work for me, if folks wanted to test.

No docs yet, so the best example is probably the new integration test test/integration/test_embedded_vault.yml

Not integrated with 'ansible-vault edit' yet, so getting the vault blobs into a playbook is a cut and paste affair. And a caveat there... most editors will add a new line to the plaintext on save, which the normal 'ansible-vault encrypt' will include and may not be what you mean if you try to encrypt something like

``` yaml
vars:
      my_password: 'hunter42'
```

If you aren't careful with the plaintext file, it will get the newline and end up with the value with a newline like:

``` yaml
vars:
     my_password: 'hunter42\n'
```

**Comment by alikins** [comment]
Ready for review.

I'll leave the commits unsquashed for now, but can start squashing them into logical chunks if needed.

File: lib/ansible/parsing/vault/__init__.py [hunk 1]
```diff
@@ -473,6 +545,7 @@ def _editor_shell_command(self, filename):
 
         return editor
 
+# TODO: does anything use this?
 class VaultFile(object):
```
Comment (bcoca) [inline lib/ansible/parsing/vault/__init__.py]: no, this was a 'prototype' i was working on, feel free to remove

**Comment by bcoca** [comment]
so most of the changes in the PR seem to be reformating/renaming of vars and code, I'm not against the changes as they clarify what a lot of the vars are, but I would like to keep these changes separate from the actual feature. It is hard to review the feature it self when it is lost in a sea of unrelated changes.

**Pull request merged**